typegraph/typegraph/types.py

```diff
@@ -504,10 +504,18 @@ class struct(typedef):
     def additional(self, t: Union[bool, TypeNode]):
         return self.replace(additional_props=t)
 
+    # creates a duplicate type with no runtime
+    def detach(self):
+        return self.replace(
+            name=f"{self.type}_{self.graph.next_type_id()}", runtime=None
+        )
+
     def compose(self, props: Dict[str, typedef]):
         new_props = dict(self.props)
         new_props.update(props)
-        return self.replace(props=frozendict(new_props))
+        return self.replace(
```

Comment:
I think `self.replace` should always replace the name...

Reply:
I don't understand, if it should always replace the name when used then isn't this already ok ? Or will it do that automatically without having to explicitly give a new name ?